judge/models/problem.py

```diff
@@ -449,6 +462,31 @@ def save(self, *args, **kwargs):
 
     save.alters_data = True
 
+    def solved_by(self, user):
+        # Return true if a full AC submission to the problem from the user exists.
+        return self.submission_set.filter(user=user.profile, result='AC', points__gte=F('problem__points')).exists()
+
+    def get_vote_perm(self, user):
```

Comment:
```suggestion
    def vote_permission_for_user(self, user):
```

Nit, spell out `perm` here because the enum is `VotePermission` and not `VotePerm`.